kolibri/plugins/learn/assets/src/composables/useDownloadRequests.js

```diff
@@ -140,15 +96,23 @@ export default function useDownloadRequests(store) {
   }
 
   function removeDownloadRequest(content) {
-    console.log(`requested removal of ${content.id}`);
-    Vue.delete(downloadRequestMap.downloads, content.id);
+    ContentRequestResource.deleteModel({
+      id: content.id,
+      contentnode_id: content.contentnode_id,
+    })
+      .then(Vue.delete(downloadRequestMap, content.id))
+      .then(set(downloadRequestMap, 'totalDownloads', Object.keys(downloadRequestMap).length));
     return Promise.resolve();
   }
 
   function removeDownloadsRequest(contentList) {
-    console.log(`requested removal of ${contentList.length} items`);
     contentList.forEach(content => {
-      Vue.delete(downloadRequestMap.downloads, content.id);
+      ContentRequestResource.deleteModel({
+        id: content.id,
+        contentnode_id: content.contentnode_id,
+      })
+        .then(Vue.delete(downloadRequestMap, content.id))
+        .then(set(downloadRequestMap, 'totalDownloads', Object.keys(downloadRequestMap).length));
     });
```

Comment:
Overall, I think it would be ideal if instead of having a separate `totalDownloads` value, we just had a computed value that counts the length of downloadRequestMap keys. However, obviously, issues with key deletion and removal causing reactive recalculation apply, but it looks like you are using Vue.delete here, so that should be ok.